clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -55,10 +57,12 @@
 
     private AtomicBoolean wakeup;
 
+    private AtomicBoolean rebalanced = new AtomicBoolean(false);
```

Comment:
This boolean is only used for detecting the first rebalance. In that case we should not invoke the invoke method.